src/main/java/com/webcrawling/service/webCrawler/impl/WebCrawlerServiceImpl.java

```diff
@@ -31,8 +32,10 @@ public class WebCrawlerServiceImpl implements WebCrawlerService {
     WebPageRepository webPageRepository;
     @Autowired
     private ExceptionRecordRepository exceptionRecordRepository;
-        public void  crawlWebsite(String startingUrl, int maxDepth, int maxDocuments) {
+    @Async
+    public void  crawlWebsite(String startingUrl, int maxDepth, int maxDocuments) {
        logger.info("Web Crawling over given URL has been started");
+       logger.info(Thread.currentThread().getName());
```

Comment:
Idea to introduce multi threading here is to process the url parallel means each thread will fetch data independently .
Do you think that now this behavior is achieved using @Async annotation. How you expect to process different url at the same time .You need to understand the executor service and it working cycle.
https://jenkov.com/tutorials/java-util-concurrent/executorservice.html